activesupport/test/core_ext/module/attribute_accessor_per_thread_test.rb

```diff
@@ -2,53 +2,56 @@
 require 'active_support/core_ext/module/attribute_accessors_per_thread'
 
 class ModuleAttributeAccessorPerThreadTest < ActiveSupport::TestCase
-  def setup
-    @class = Class.new do
-      thread_mattr_accessor :foo
-      thread_mattr_accessor :bar,  instance_writer: false
-      thread_mattr_reader   :shaq, instance_reader: false
-      thread_mattr_accessor :camp, instance_accessor: false
-    end
+  class Foo
```

Comment:
Anonymous classes don't work with `thread_mattr_accessor`. 
`thread_mattr_accessor` uses class name as part of thread local variable name.

Following code set `Thread[:attr_Account_user]`.

```
class Account
  thread_mattr_accessor :user
end
```

But anonymous class set `Thread[:attr_user]`.  Its subclass use same variable because
both of them don't have name.